src/plugins/extra/heading_anchors.rs

```diff
@@ -0,0 +1,148 @@
+use slug::slugify;
+
+use crate::{
+    parser::{core::CoreRule, inline::Text},
+    plugins::cmark::{
+        block::heading::ATXHeading,
+        inline::{
+            backticks::CodeInline,
+            emphasis::{Em, Strong},
+            link::Link,
+        },
+    },
+    MarkdownIt, Node, NodeValue, Renderer,
+};
+
+use super::strikethrough::Strikethrough;
+
+#[derive(Debug)]
+struct HeadingWithSlug {
+    level: u8,
+    slug: String,
+}
+
+impl NodeValue for HeadingWithSlug {
+    fn render(&self, node: &Node, fmt: &mut dyn Renderer) {
+        static TAG: [&str; 6] = ["h1", "h2", "h3", "h4", "h5", "h6"];
+        debug_assert!(self.level >= 1 && self.level <= 6);
+
+        let tag = TAG[self.level as usize - 1];
+        let attrs = &[("id", self.slug.clone())];
+
+        fmt.cr();
+        fmt.open(tag, attrs);
+        fmt.contents(&node.children);
+        fmt.close(tag);
+        fmt.cr();
+    }
+}
+
+struct HeadingsWithIds;
+
+impl CoreRule for HeadingsWithIds {
+    fn run(node: &mut Node, _: &MarkdownIt) {
+        // Walk the existing AST and replace all headings with our custom headings
+        node.walk_mut(|node, _| {
+            if let Some(heading) = node.cast::<ATXHeading>() {
```

Comment:
Setext heading is missing.

There are two styles in markdown:

```md
## atx heading

setext heading
==============
```

This rule only covers first, but I guess both needs to be present?

It was my questionable design decision to make separate nodes for both types of headings. If we are to follow it, this code has to be duplicated with `ATXHeadingWithSlug` and `SetextHeadingWithSlug`. *(Or should I make a large change and unify those tokens?)*